src/probnum/diffeq/__init__.py

```diff
@@ -33,9 +24,6 @@
     "fitzhughnagumo",
     "lotkavolterra",
     "seir",
-    "rigidbody",
-    "vanderpol",
-    "threebody",
```

Comment:
I moved those three to `probnum.problems.zoo`, because (i) vanderpol and threebody are used in doctests to show a comparison between initialisation with RK and initialisation with TM; (ii) rigidbody has a jacobian, which helps with setting up the unit test structure in the problem module. Neither of those three was used anywhere else before, so no functionality is lost.